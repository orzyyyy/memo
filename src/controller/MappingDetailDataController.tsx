import React from 'react';
import MappingDetail from '../pages/MappingDetail';
import { DataSource } from 'mini-xmind/lib/canvas/core';
import { useFetchDocumentData } from '../hooks/useFetchDocumentData';
import { MarkdownEditorSaveProps, showMessageAfterFetching } from './MarkdownEditorDataController';
import { getPathNameFromUrl } from '../utils';

const MappingDetailDataController = () => {
  const id: string = getPathNameFromUrl();

  const handleOnSave = async (newData: DataSource) => {
    const params: MarkdownEditorSaveProps = {
      layout: newData,
      id,
      category: 'mapping',
    };
    const response = await fetch('/api/memo/document/update/content', {
      method: 'POST',
      body: JSON.stringify(params),
      mode: 'cors',
      headers: { 'Content-Type': 'application/json' },
    });
    const result = await response.text();
    showMessageAfterFetching(result);
  };

  let [data, setData] = useFetchDocumentData(id, 'mapping', handleOnSave);
  if (!data) {
    data = {
      position: {
        root: {
          x: -3000,
          y: -3000,
        },
      },
      block: {},
      tag: {},
      line: {},
    };
  }

  const handleOnChange = (data: DataSource) => {
    setData(data);
  };

  return <MappingDetail dataSource={data} onChange={handleOnChange} />;
};

export default MappingDetailDataController;
