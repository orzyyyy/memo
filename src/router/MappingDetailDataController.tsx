import React, { Suspense, lazy } from 'react';
import ReactDOM from 'react-dom';
import TohoLoading from '../pages/TohoLoading';

const MappingDetailDataController = lazy(() => import('../controller/MappingDetailDataController'));

ReactDOM.render(
  <Suspense fallback={<TohoLoading />}>
    <MappingDetailDataController />
  </Suspense>,
  document.getElementById('root'),
);
