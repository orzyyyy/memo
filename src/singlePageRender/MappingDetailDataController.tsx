import React from 'react';
import ReactDOM from 'react-dom';
import MappingDetailDataController from '../controller/MappingDetailDataController';

ReactDOM.render(
  <MappingDetailDataController />,
  document.getElementById('root'),
);
