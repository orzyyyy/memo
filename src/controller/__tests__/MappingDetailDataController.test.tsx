import React from 'react';
import { mount } from 'enzyme';
import MappingDetailDataController from '../MappingDetailDataController';
import fetchMock from 'fetch-mock';

describe('MappingDetailDataController', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  beforeEach(() => {
    fetchMock.mock('/document/update', {});
    fetchMock.mock('/mapping//.json', { test: 1 });
  });

  afterAll(() => {
    errorSpy.mockRestore();
  });

  afterEach(() => {
    fetchMock.restore();
  });

  it('handleOnSave', async () => {
    const wrapper = mount(<MappingDetailDataController />);
    await wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
