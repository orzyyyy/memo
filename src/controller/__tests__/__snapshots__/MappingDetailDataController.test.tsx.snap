// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MappingDetailDataController handleOnSave 1`] = `
<MappingDetailDataController>
  <MappingDetail
    dataSource={
      Object {
        "block": Object {},
        "line": Object {},
        "position": Object {
          "root": Object {
            "x": -3000,
            "y": -3000,
          },
        },
        "tag": Object {},
      }
    }
    onChange={[Function]}
  >
    <div
      className="mapping"
    >
      <Toolbar>
        <div
          className="Toolbar"
        >
          <ul>
            <li>
              <div
                className="icon-minus"
              />
            </li>
            <li>
              <div
                className="icon-branches"
              />
            </li>
            <li
              draggable={true}
              key="toolbar-block"
              onDragStart={[Function]}
            >
              <div
                className="icon-border"
              />
            </li>
            <li
              draggable={true}
              key="toolbar-tag"
              onDragStart={[Function]}
            >
              <div
                className="icon-tag"
              />
            </li>
          </ul>
        </div>
      </Toolbar>
      <Canvas
        className="canvas-wrapper"
        data={
          Object {
            "block": Object {},
            "line": Object {},
            "position": Object {
              "root": Object {
                "x": -3000,
                "y": -3000,
              },
            },
            "tag": Object {},
          }
        }
        onChange={[Function]}
      >
        <Draggable
          allowAnyClick={false}
          axis="both"
          bounds={false}
          cancel={null}
          defaultClassName="react-draggable"
          defaultClassNameDragged="react-draggable-dragged"
          defaultClassNameDragging="react-draggable-dragging"
          defaultPosition={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          disabled={false}
          enableUserSelectHack={true}
          grid={null}
          handle={null}
          offsetParent={null}
          onDrag={[Function]}
          onMouseDown={[Function]}
          onStart={[Function]}
          onStop={[Function]}
          position={
            Object {
              "x": -3000,
              "y": -3000,
            }
          }
          scale={1}
          transform={null}
        >
          <DraggableCore
            allowAnyClick={false}
            cancel={null}
            disabled={false}
            enableUserSelectHack={true}
            grid={null}
            handle={null}
            offsetParent={null}
            onDrag={[Function]}
            onMouseDown={[Function]}
            onStart={[Function]}
            onStop={[Function]}
            scale={1}
            transform={null}
          >
            <div
              className="Canvas canvas-wrapper react-draggable"
              onDragOver={[Function]}
              onDrop={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onWheel={[Function]}
              style={
                Object {
                  "transform": "translate(-3000px,-3000px)",
                }
              }
              transform={null}
            >
              <BlockGroup
                data={Object {}}
                lineData={Object {}}
                offset={
                  Object {
                    "x": -3000,
                    "y": -3000,
                  }
                }
                onChange={[Function]}
                onContextMenu={[Function]}
                onWheel={[Function]}
              />
              <TagGroup
                data={Object {}}
                lineData={Object {}}
                onChange={[Function]}
                onContextMenu={[Function]}
                onWheel={[Function]}
              />
              <LineGroup
                data={Object {}}
                offset={
                  Object {
                    "x": -3000,
                    "y": -3000,
                  }
                }
                orientation="horizonal"
              />
            </div>
          </DraggableCore>
        </Draggable>
      </Canvas>
    </div>
  </MappingDetail>
</MappingDetailDataController>
`;
